pkg/portallocator/portallocator_test.go

```diff
@@ -48,7 +48,7 @@ func TestPortAllocatorAllocate(t *testing.T) {
 
 	t.Run("test allocated port counts", func(t *testing.T) {
 		m := agtesting.NewMocks()
-		pa := NewPortAllocator(10, 50, m.KubeInformerFactory, m.AgonesInformerFactory)
+		pa := New(10, 50, m.KubeInformerFactory, m.AgonesInformerFactory).(*portAllocator)
```

Comment:
nit: It seems like we should be able to avoid casting here. What about a `new` (non-exported) method that returns `*portAllocator` that is called by `New` (which then just wraps the return value into the exported `portAllocator.Interface`)?

Reply:
Sounds good, done.